### PR TITLE
Add property builder helper

### PR DESCRIPTION
**Pull request by Evangelink**
Follow-up of #23720

### CODE REVIEW

File: src/Cli/dotnet/CommonOptions.cs [hunk 1]
```diff
@@ -272,6 +272,41 @@ private static IEnumerable<string> ForwardSelfContainedOptions(bool isSelfContai
             parseResult.HasOption(ArchitectureOption) ||
             parseResult.HasOption(LongFormArchitectureOption) ||
             parseResult.HasOption(OperatingSystemOption);
+
+        internal static string BuildProperty<TValue>(string name, TValue value, bool escapeValue = false)
```
Comment (MarcoRossignoli) [inline src/Cli/dotnet/CommonOptions.cs:276]: escape or escaped?

Comment (Evangelink) [reply src/Cli/dotnet/CommonOptions.cs:276]: I would say `escape` because that's what we want to do not what's done before.

Comment (Evangelink) [reply src/Cli/dotnet/CommonOptions.cs:276]: Thanks! There was indeed a problem between what I was saying and the implemented condition!

**Comment by Nirmal4G** [comment]
We could also do the same thing to `-t` targets options.

Related Issue to look at: #26176

**Comment by Evangelink** [comment]
Damn totally forgot about this PR, let me try to take it back on track!

**Comment by Nirmal4G** [comment]
You can look for your issues and PRs on GitHub through here: https://github.com/issues and https://github.com/pulls

Same is also available as a link in the header in almost every GitHub page beside the search box! You can track your open source adventures there!! 😇🙃

File: src/Cli/dotnet/CommonOptions.cs [hunk 2]
```diff
@@ -272,6 +272,42 @@ private static IEnumerable<string> ForwardSelfContainedOptions(bool isSelfContai
...
+        {
+            var valueString = value.ToString();
+
+            var needsToBeEscaped = escapeValue
+                && !valueString.StartsWith("\"", StringComparison.Ordinal)
+                && !valueString.EndsWith("\"", StringComparison.Ordinal);
+
+            return needsToBeEscaped
+                ? $"-property:{name}=\"{valueString}\""
```
Comment (baronfel) [inline src/Cli/dotnet/CommonOptions.cs:285]: minor nit - for user-supplied properties we've standardized on forwarding those to MSBuild as `--property:<VALUE>`, could this helper do the same? just to make sure it all looks uniform.

**Comment by marcpopMSFT** [comment]
Old PR review: This PR is more than a year old and does not appear to have active engagement. A reviewer has been notified offline to review and take action. If no action is taken, this PR will be closed in 14 days. Please comment if you want this PR left open for further investigation.

**Comment by Evangelink** [comment]
Closing as I didn't have enough time to complete and it's now relatively obsolete.

**Pull request closed without merging**